src/NuGet.Core/NuGet.PackageManagement/Utility/PackageGraphAnalysisUtilities.cs

```diff
@@ -0,0 +1,123 @@
+        public static Func<IPackageWithDependants, bool> TransitivePackagesPredicate = dependencyItem => dependencyItem.DependingPackages.Any();
+
+        /// <summary>
+        /// Returns package dependency info for the given package identities in the given resource. It returns null if any protocol errors occur.
+        /// For example, the feed is not accessible.
+        /// </summary>
+        /// <param name="packageIdentities">A collection of <see cref="PackageIdentity"/> to get info for.</param>
+        /// <param name="nuGetFramework">framework for determining the dependency groups of packages</param>
+        /// <param name="dependencyInfoResource">The resource to fetch dependency info from. Could be http/file feed/global packages folder/solution packages folder.</param>
+        /// <param name="sourceCacheContext">Caching context. Only reallly applicable when the dependency info resource is http based</param>
+        /// <param name="includeUnresolved">Whether to include unresolved packages in the list. If true, the unresolved packages will have an empty dependencies collection.</param>
+        /// <param name="logger">logger</param>
+        /// <param name="cancellationToken">cancellation token</param>
+        /// <returns>A collection of <see cref="PackageDependencyInfo"/>, null if a protocol exception happens.  </returns>
+        public static async Task<IEnumerable<PackageDependencyInfo>> GetDependencyInfoForPackageIdentitiesAsync(IEnumerable<PackageIdentity> packageIdentities,
+            NuGetFramework nuGetFramework,
+            DependencyInfoResource dependencyInfoResource,
+            SourceCacheContext sourceCacheContext,
+            bool includeUnresolved,
+            ILogger logger,
+            CancellationToken cancellationToken)
+        {
+            try
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var results = new HashSet<PackageDependencyInfo>(PackageIdentity.Comparer);
+                foreach (var package in packageIdentities)
+                {
+                    var packageDependencyInfo = await dependencyInfoResource.ResolvePackage(
+                        package,
+                        nuGetFramework,
+                        sourceCacheContext,
+                        logger,
+                        cancellationToken);
+
+                    if (packageDependencyInfo != null)
+                    {
+                        results.Add(packageDependencyInfo);
+                    }
+                    else if (includeUnresolved)
+                    {
+                        results.Add(new PackageDependencyInfo(package, null));
+                    }
+                }
+
+                return results;
+            }
+            catch (NuGetProtocolException)
+            {
+                return null;
```

Comment:
just curious, don't we need to log the error message before returning null ?

Reply:
Good point, we can log it. 
Originally my thought was, let's not change the experience in the migrator, but this would be a good improvement.